### PR TITLE
bugFix

### PR DESCRIPTION
**Pull request by jerometujague**
I'm running into an unhandled promise rejection when clicking on the cancel appointment button. Can you take a look and see if you can figure out what I've done wrong?

### CODE REVIEW

File: src/main/java/com/commercebank/controller/AppointmentController.java [hunk 1]
```diff
@@ -60,6 +60,11 @@ public AppointmentController(AppointmentDAO appointmentDAO, CustomerDAO customer
         return appointmentDAO.list();
     }
 
+    @RequestMapping(value = "/delete", method = RequestMethod.DELETE)
```
Comment (jmbockhorst) [inline src/main/java/com/commercebank/controller/AppointmentController.java]: Pass the id in the URL, like `/delete/{id}`, then you don't need the extra code in the DAO.

File: src/main/java/com/commercebank/dao/AppointmentDAO.java [hunk 2]
```diff
@@ -43,6 +43,16 @@
         return appointments;
     }
 
+    public int findId(){
```
Comment (jmbockhorst) [inline src/main/java/com/commercebank/dao/AppointmentDAO.java]: Not needed.

File: src/main/resources/assets/src/detailsPage.js [hunk 3]
```diff
@@ -9,6 +9,22 @@ export default class Details extends React.Component {
...
+        this.props.setStateValue('loading', true);
+        // Send the delete request
+        await $.ajax({
+            type: 'DELETE',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            url: '/api/appointments/delete',
+            data: JSON.stringify({
```
Comment (jmbockhorst) [inline src/main/resources/assets/src/detailsPage.js]: You need to remove the `data` field.

File: src/main/resources/assets/src/detailsPage.js [hunk 4]
```diff
@@ -9,6 +9,22 @@ export default class Details extends React.Component {
...
+    async cancelAppointment(){
+        this.props.setStateValue('loading', true);
+        // Send the delete request
+        await $.ajax({
+            type: 'DELETE',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            url: '/api/appointments/delete',
```
Comment (jmbockhorst) [inline src/main/resources/assets/src/detailsPage.js]: Pass the appointment id to the url: `'/api/appointments/delete/' + this.props.appointment.id`

File: src/main/java/com/commercebank/controller/AppointmentController.java [hunk 5]
```diff
@@ -60,6 +60,11 @@ public AppointmentController(AppointmentDAO appointmentDAO, CustomerDAO customer
...
+    void deleteAppointment(){
```
Comment (jmbockhorst) [inline src/main/java/com/commercebank/controller/AppointmentController.java]: And then get the path variable: `deleteAppoinmtment(@PathVariable int id)`

**Comment by jerometujague** [comment]
I made the changed you requested, but the server is still crashing when I click the schedule button.

**Pull request closed without merging**